src/mongocxx/client.hpp

```diff
@@ -61,7 +68,7 @@ class MONGOCXX_API client {
     ///   Additional options that cannot be specified via the mongodb_uri
     ///
     client(
-        const class uri& mongodb_uri = mongocxx::uri(),
+        const class uri& mongodb_uri,
```

Comment:
My preference would be to keep the default constructor for URI. I think you have a good point about non-default uri's being used almost everywhere, so I think this makes sense.